pystac/catalog.py

```diff
@@ -665,26 +665,22 @@ def generate_subcatalogs(
         for link in item_links:
             link.resolve_stac_object(root=self.get_root())
             item = cast(pystac.Item, link.target)
-            item_parts = layout_template.get_template_values(item)
+            subcat_ids = layout_template.substitute(item).split("/")
```

Comment:
Thanks for submitting this @fnattino! It looks like there is a test failure on Windows that I think is probably due to splitting on `"/"`, which wouldn't be used as a filepath delimiter on Windows:

```
FAIL: test_generate_subcatalogs_works_for_subcatalogs_with_same_ids (test_catalog.CatalogTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\pystac\pystac\tests\test_catalog.py", line 541, in test_generate_subcatalogs_works_for_subcatalogs_with_same_ids
    self.assertEqual(len(result), 6)
AssertionError: 4 != 6

----------------------------------------------------------------------
Ran 527 tests in 52.455s

FAILED (failures=1, skipped=3)
```

I'm not as familiar with the templating code in PySTAC, but I'll take a look later today to see if there is a good fix for this.

Reply:
Uh, thanks for the heads-up! Indeed the template used in one of the tests is  constructed via the `join_path_or_url` function:
https://github.com/stac-utils/pystac/blob/e744897c63fd5561040396231f52e1cec398b024/tests/test_catalog.py#L515
which results in `"\"` being used as separator on Windows (and the test failing in the splitting).

A question: Are users on Windows actually expected to provide the template using `"\"` as separator? Since the template is just a string used to define the catalog structure, would it make sense to adopt the convention of using one separator (say `"/"`) on all OSs? If you agree, I would be happy to fix the test and add a line to the documentation..

Reply:
> Since the template is just a string used to define the catalog structure, would it make sense to adopt the convention of using one separator (say `"/"`) on all OSs?

I think the safest bet might actually be to use `os.sep` instead of `"/"` so that it uses whatever path separator is set for that system.

```suggestion
            subcat_ids = layout_template.substitute(item).split(os.sep)
```

I'm a little fuzzy on whether you can use `/` as a path separator on Windows, and I know that in rare cases users can also set a different path separator for their system (but I don't know if `/` will also still work in that case). If we were to start using `/` as our file path separator we might also need to update the `LayoutTemplate.substitute` code to replace `"/"` with `os.sep` in order to get actual sub-directories (again, I'm a little fuzzy on how exactly this works on Windows), which would cause problems when templating URLs on Windows.

@gadomski @matthewhanson Do either of you have a strong opinion on how this should work?

Reply:
My instinct is to always use `/` when creating paths, templates, etc, and only convert to OS paths when needed to do filesystem operations. IMO using [PurePath](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath) would be the cleanest abstraction but would likely require some significant refactoring.

Reply:
> My instinct is to always use `/` when creating paths

Yeah, mine too. And actually now that I look at assertions [this this](https://github.com/stac-utils/pystac/blob/e744897c63fd5561040396231f52e1cec398b024/tests/test_layout.py#L46) it seems like we are implicitly assuming that we're using a `/` separator anyway.

@fnattino Let's stick with your suggestion of using `"/"` as the separator, updating the test and adding a note to the docs.

Reply:
Just implemented the changes, sorry for the delay with this!

Reply:
No problem, thanks for this!